Modules/_struct.c

```diff
@@ -2503,6 +2503,9 @@ _structmodule_exec(PyObject *m)
     if (PyModule_AddType(m, (PyTypeObject *)state->PyStructType) < 0) {
         return -1;
     }
+    if (_PyImmutability_RegisterFreezable((PyTypeObject *)state->PyStructType) < 0){
+        return -1;
+    }
```

Comment:
The file `Modules/_elementtree.c` uses `_PyImport_GetModuleAttrString("immutable", "register_freezable");` to get the `register_freezable` function or handle failure if the "immutable" module is not available. This one uses the builtin function `_PyImmutability_RegisterFreezable` from pycore. How is it decided which function is used?